aml_module/invoker.py

```diff
@@ -0,0 +1,167 @@
+import argparse
+from urllib.parse import unquote
+import pandas as pd
+import numpy as np
+from azureml.studio.core.logger import module_logger as logger
+from azureml.studio.core.io.data_frame_directory import load_data_frame_from_directory, save_data_frame_to_directory
+from azureml.studio.core.utils.column_selection import ColumnSelection
+from azureml.studio.internal.error_handler import error_handler
+import sr_detector
+from error_messages import *
+from constants import *
+
+PACKAGE_NAME = 'spectral_residual_anomaly_detection_module'
+VERSION = '0.0.1'
+
+
+def str2bool(v):
+    if isinstance(v, bool):
+        return v
+    if v.lower() in ('yes', 'true', 't', 'y', '1'):
+        return True
+    elif v.lower() in ('no', 'false', 'f', 'n', '0'):
+        return False
+    else:
+        raise argparse.ArgumentTypeError('Boolean value expected.')
+
+
+def is_timestamp_ascending(timestamps):
+    count = len(timestamps)
+
+    for i in range(count - 1):
+        if timestamps[i] > timestamps[i + 1]:
+            return -1
+        elif timestamps[i] == timestamps[i + 1]:
+            return -2
+    return 0
+
+
+@error_handler
```

Comment:
It would be better to wrap the whole main() function to make sure all exceptions could be shown in Designer